assets/pages/new-contributions-landing/components/StripePaymentRequestButton.jsx

```diff
@@ -0,0 +1,242 @@
+import type { IsoCurrency } from 'helpers/internationalisation/currency';
+import { getStripeKey } from 'helpers/paymentIntegrations/newPaymentFlow/stripeCheckout';
+import { getAmount } from 'helpers/contributions';
+import type { ContributionType, OtherAmounts, SelectedAmounts } from 'helpers/contributions';
+import type { PaymentAuthorisation } from 'helpers/paymentIntegrations/newPaymentFlow/readerRevenueApis';
+import { checkAmountOrOtherAmount, isValidEmail } from 'helpers/formValidation';
+import { type PaymentResult } from 'helpers/paymentIntegrations/newPaymentFlow/readerRevenueApis';
+import { isInStripePaymentRequestAllowedCountries } from 'helpers/internationalisation/country';
+import type { ApplePayTestVariant } from 'helpers/abTests/abtestDefinitions';
+import type { CountryGroupId } from 'helpers/internationalisation/countryGroup';
+import type { IsoCountry } from 'helpers/internationalisation/country';
+import { hiddenIf } from 'helpers/utilities';
+import { logException } from 'helpers/logger';
+import type { State } from '../contributionsLandingReducer';
+import {
+  setCanMakeApplePayPayment,
+  setStripePaymentRequestButtonClicked,
+  setPaymentRequest,
+  onStripePaymentRequestApiPaymentAuthorised,
+  updateEmail,
+} from '../contributionsLandingActions';
+
+
+// ----- Types -----//
+
+/* eslint-disable react/no-unused-prop-types */
+type PropTypes = {|
+  country: IsoCountry,
+  currency: IsoCurrency,
+  isTestUser: boolean,
+  stripeCheckout: Object | null,
+  contributionType: ContributionType,
+  applePayTestVariant: ApplePayTestVariant,
+|};
+
+const mapStateToProps = (state: State) => ({
+  selectedAmounts: state.page.form.selectedAmounts,
+  otherAmounts: state.page.form.formData.otherAmounts,
+  canMakeApplePayPayment: state.page.form.stripePaymentRequestButtonData.canMakeApplePayPayment,
+  paymentRequest: state.page.form.stripePaymentRequestButtonData.paymentRequest,
+  countryGroupId: state.common.internationalisation.countryGroupId,
+});
+
+const mapDispatchToProps = (dispatch: Function) => ({
+  onPaymentAuthorised:
+    (token, complete) => { dispatch(onStripePaymentRequestApiPaymentAuthorised(token, complete)); },
+  setCanMakeApplePayPayment:
+    (canMakeApplePayPayment) => { dispatch(setCanMakeApplePayPayment(canMakeApplePayPayment)); },
+  setPaymentRequest:
+    (paymentRequest) => { dispatch(setPaymentRequest(paymentRequest)); },
+  updateEmail: (email: string) => { dispatch(updateEmail(email)); },
+  setStripePaymentRequestButtonClicked: () => { dispatch(setStripePaymentRequestButtonClicked()); },
+});
+
+
+function updateUserEmail(data: Object, setEmail: string => void) {
+  const email = data.payerEmail;
+  if (email && isValidEmail(email)) {
+    setEmail(email);
+  } else {
+    const msg = email
+      ? `Failed to set email for stripe payment request user with email: ${email}`
+      : 'Failed to set email: no email in data object';
+    logException(msg);
+  }
```

Comment:
we are checking `email` is false twice, why not only check once, something like this
```
if (email) {
  if (isValidEmail(email)) {
    setEmail(email);
  } else {
    logException('Failed to set email for stripe payment request user with email: ${email}');
  }
} else {
  logException('Failed to set email: no email in data object');
}
```

Reply:
👍